compiler/src/test/java/dagger/internal/codegen/ComponentProcessorTest.java

```diff
@@ -1837,4 +1837,274 @@ public void genericTestToLetMeDebugInEclipse() {
+        "    this.bufferProvider = ComponentWithEnclosedType$Buffer_Factory.create(internalProvider);",
+        "  }",
+        "",
+        "  @Override",
+        "  public Buffer buffer() {",
+        "    return bufferProvider.get();",
+        "  }",
+        "",
+        "  public static final class Builder {",
+        "    private Module module;",
+        "  ",
+        "    private Builder() {",
+        "    }",
+        "  ",
+        "    public ComponentWithEnclosedType build() {",
+        "      if (module == null) {",
+        "        this.module = new Module();",
+        "      }",
+        "      return new DaggerComponentWithEnclosedType(this);",
+        "    }",
+        "  ",
+        "    public Builder module(Module module) {",
+        "      if (module == null) {",
+        "        throw new NullPointerException(\"module\");",
+        "      }",
+        "      this.module = module;",
+        "      return this;",
+        "    }",
+        "  }",
+        "}");
+    JavaFileObject generatedInternalTypeFactory = JavaFileObjects.forSourceLines("test.Module_InternalFactory",
+        "package test;",
+        "",
+        "import dagger.internal.Factory;",
+        "import java.nio.Buffer;",
+        "import javax.annotation.Generated;",
+        "",
+        "@Generated(\"dagger.internal.codegen.ComponentProcessor\")",
+        "public final class Module_InternalFactory implements Factory<Buffer> {",
+        "  private final Module module;",
+        "",
+        "  public Module_InternalFactory(Module module) {",
+        "    assert module != null;",
+        "    this.module = module;",
+        "  }",
+        "",
+        "  @Override",
+        "  public Buffer get() {", // No ambiguity in this file.
+        "    Buffer provided = module.internal();",
+        "    if (provided == null) {",
+        "      throw new NullPointerException(\"Cannot return null from a non-@Nullable @Provides method\");",
+        "    }",
+        "    return provided;",
+        "  }",
+        "",
+        "  public static Factory<Buffer> create(Module module) {",
+        "    return new Module_InternalFactory(module);",
+        "  }",
+        "}");
+    JavaFileObject generatedEnclosedTypeFactory = JavaFileObjects.forSourceLines("test.ComponentWithEnclosedType$Buffer_Factory",
+        "package test;",
+        "",
+        "import dagger.internal.Factory;",
+        "import java.nio.Buffer;",
+        "import javax.annotation.Generated;",
+        "import javax.inject.Provider;",
+        "import test.ComponentWithEnclosedType;",
+        "",
+        "@Generated(\"dagger.internal.codegen.ComponentProcessor\")",
+        "public final class ComponentWithEnclosedType$Buffer_Factory implements Factory<ComponentWithEnclosedType.Buffer> {",
+        "  private final Provider<Buffer> bProvider;",
+        "",
+        "  public ComponentWithEnclosedType$Buffer_Factory(Provider<Buffer> bProvider) {",
```

Comment:
curious.. are we intentionally still having these ctors public eventhough there's a factory?